internal/transform/effects.go

```diff
@@ -0,0 +1,1289 @@
+		EffectSignerCreated,
+		map[string]interface{}{
+			"public_key": op.Destination.Address(),
+			"weight":     keypair.DefaultSignerWeight,
+		},
+	)
+}
+
+func (e *effectsWrapper) addPaymentEffects() {
+	op := e.operation.operation.Body.MustPaymentOp()
+
+	details := map[string]interface{}{"amount": amount.String(op.Amount)}
+	addAssetDetails(details, op.Asset, "")
+
+	e.addMuxed(
+		&op.Destination,
+		EffectAccountCredited,
+		details,
+	)
+	e.addMuxed(
+		e.operation.SourceAccount(),
+		EffectAccountDebited,
+		details,
+	)
+}
+
+func (e *effectsWrapper) pathPaymentStrictReceiveEffects() error {
+	op := e.operation.operation.Body.MustPathPaymentStrictReceiveOp()
+	resultSuccess := e.operation.OperationResult().MustPathPaymentStrictReceiveResult().MustSuccess()
+	source := e.operation.SourceAccount()
+
+	details := map[string]interface{}{"amount": amount.String(op.DestAmount)}
+	addAssetDetails(details, op.DestAsset, "")
+
+	e.addMuxed(
+		&op.Destination,
+		EffectAccountCredited,
+		details,
+	)
+
+	result := e.operation.OperationResult().MustPathPaymentStrictReceiveResult()
+	details = map[string]interface{}{"amount": amount.String(result.SendAmount())}
+	addAssetDetails(details, op.SendAsset, "")
+
+	e.addMuxed(
+		source,
+		EffectAccountDebited,
+		details,
+	)
+
+	return e.addIngestTradeEffects(*source, resultSuccess.Offers, false)
+}
+
+func (e *effectsWrapper) addPathPaymentStrictSendEffects() error {
+	source := e.operation.SourceAccount()
+	op := e.operation.operation.Body.MustPathPaymentStrictSendOp()
+	resultSuccess := e.operation.OperationResult().MustPathPaymentStrictSendResult().MustSuccess()
+	result := e.operation.OperationResult().MustPathPaymentStrictSendResult()
+
+	details := map[string]interface{}{"amount": amount.String(result.DestAmount())}
+	addAssetDetails(details, op.DestAsset, "")
+	e.addMuxed(&op.Destination, EffectAccountCredited, details)
+
+	details = map[string]interface{}{"amount": amount.String(op.SendAmount)}
+	addAssetDetails(details, op.SendAsset, "")
+	e.addMuxed(source, EffectAccountDebited, details)
+
+	return e.addIngestTradeEffects(*source, resultSuccess.Offers, true)
+}
+
+func (e *effectsWrapper) addManageSellOfferEffects() error {
+	source := e.operation.SourceAccount()
+	result := e.operation.OperationResult().MustManageSellOfferResult().MustSuccess()
+	return e.addIngestTradeEffects(*source, result.OffersClaimed, false)
+}
+
+func (e *effectsWrapper) addManageBuyOfferEffects() error {
+	source := e.operation.SourceAccount()
+	result := e.operation.OperationResult().MustManageBuyOfferResult().MustSuccess()
+	return e.addIngestTradeEffects(*source, result.OffersClaimed, false)
+}
+
+func (e *effectsWrapper) addCreatePassiveSellOfferEffect() error {
+	result := e.operation.OperationResult()
+	source := e.operation.SourceAccount()
+
+	var claims []xdr.ClaimAtom
+
+	// KNOWN ISSUE:  stellar-core creates results for CreatePassiveOffer operations
+	// with the wrong result arm set.
```

Comment:
Can you cite an issue, code snippet, or slack thread for this? Would be nice to have a reference!

Reply:
About this and the main concern you addressed in the comment, I forgot to mention that [`transform/effects.go`](https://github.com/stellar/stellar-etl/blob/export_history_effects/internal/transform/effects.go) is virtually [`effects_processor.go`](https://github.com/stellar/go/blob/master/services/horizon/internal/ingest/processors/effects_processor.go) from Horizon.

I'll edit this PR to link files from here to those source files from Horizon.

Reply:
About adding fuzzing support, I think it's a great idea. The test file is massive and confuse.
The only problem is that we're depending on `history_effects` table to recreate the `enriched_operations` table. Maybe we should discuss about that in Slack.